identity-core/src/common/one_or_many.rs

```diff
@@ -203,6 +204,51 @@ impl<'a, T> Iterator for OneOrManyIter<'a, T> {
   }
 }
 
+// =============================================================================
+// IntoIterator
+// =============================================================================
+
+enum Either<L, R> {
+  Left(L),
+  Right(R),
+}
+
+pub struct OneOrManyIterator<T> {
```

Comment:
nit: Maybe `OneOrManyIntoIter` would be more consistent with the naming scheme from the standard library (see https://doc.rust-lang.org/std/vec/struct.IntoIter.html and https://doc.rust-lang.org/std/slice/struct.Iter.html where the later is returned from `Vec::iter`). 

Also consider adding some minimal documentation here like: 
```
/// This struct is created by the `into_iter` method on [`OneOrMany`] (provided by the [IntoIterator](https://doc.rust-lang.org/std/iter/trait.IntoIterator.html) trait).
```

Finally it would be great with a simple test that confirms that works as expected for a `OneOrMany` collection with zero, one and two (or more) elements.